File: Kattis/A_Vicious_Pikeman_Easy.cpp
#include<iostream>
#include<cstdio>
#include<algorithm>
#define ll long long

using namespace std;

const ll MOD_7 = 1000000007;

ll mod(ll a, ll b){
    return ((a%b) + b)% b;
}

int main(){
    ll N,T; scanf("%lld %lld",&N,&T);
    ll a[N];
    ll A,B,C; scanf("%lld %lld %lld %lld",&A,&B,&C,&a[0]);
    for(int i = 1; i < N; i++){
        a[i] = mod((A*a[i-1] + B),C) + 1;
    }
    sort(a,a+N,less<int>());
    ll sum=0;
    ll tm=0;
    int cnt=0;
    for(int i= 0; i < N; i++){
        if(sum+a[i] <= T){
            sum  += mod(tm + a[i],MOD_7);
            tm+=a[i];
            cnt++;
        } else
            break;
    }
    printf("%d %lld\n",cnt,sum);
    return 0;
}

File: Kattis/A_Vicious_Pikeman_Easy.py
MOD_7 = 1000000007

[N,T] = map(int,raw_input().split())
a = [0]*N
[A,B,C,t0] = map(int,raw_input().split())
a[0]=t0
for i in range(1,N):
    a[i] = ((A*a[i-1] + B) % C) + 1
a.sort()
print a
sm = 0
cnt=0
tm = 0
for i in range(N):
    if sm+a[i] <= T:
        tm+=a[i]
        sm+=tm
        cnt+=1
    else:
        break
print "%d %d" %(cnt,(sm%MOD_7))
